src-tauri/src/operations/node_operations.rs

```diff
@@ -79,85 +91,298 @@ pub fn get_nodes(state: State<'_, AppState>) -> Result<Vec<NodeInfo>> {
+                        .map_err(|e| AppError::Custom(format!("Failed to lock output: {}", e)))?;
+                    output_lock.push_str(&cleaned_line);
+                    output_lock.push('\n');
+                }
+                app_handle
+                    .emit_all(&format!("node-output-{}", node_name), cleaned_line + "\n")
+                    .map_err(|e| {
+                        AppError::Custom(format!(
+                            "Failed to emit output for node {}: {}",
+                            node_name, e
+                        ))
+                    })?;
+
+                Ok(())
+            };
+
+            for line in stdout_reader.lines().chain(stderr_reader.lines()) {
+                if let Err(e) = process_line(line) {
+                    eprintln!("Error processing line for node {}: {}", node_name, e);
+                    return Err(AppError::Custom(e.to_string()));
+                }
+            }
+            Ok(())
+        }
+    });
+
+    let node_process = NodeProcess {
+        process: Some(process),
+        stdin: Some(tx),
+        output,
+    };
+    manager.nodes.insert(node_name.clone(), node_process);
+
+    Ok(OperationResult {
+        success: true,
+        message: "Node start command issued. Check the output for status.".to_string(),
+    })
+}
+
+pub fn get_node_output(state: State<'_, AppState>, node_name: String) -> Result<OperationResult> {
+    let manager = state
+        .node_manager
+        .lock()
+        .map_err(|e| AppError::Custom(format!("Failed to lock node manager: {}", e)))?;
+
+    match manager.nodes.get(&node_name) {
+        Some(node_process) => {
+            let output = node_process
+                .output
+                .lock()
+                .map_err(|e| AppError::Custom(format!("Failed to lock output: {}", e)))?
+                .clone();
+
+            Ok(OperationResult {
+                success: true,
+                message: output,
+            })
+        }
+        None => Ok(OperationResult {
+            success: false,
+            message: format!("Node not found: {}", node_name),
+        }),
+    }
+}
+
+pub async fn stop_node_process(
+    state: State<'_, AppState>,
+    node_name: String,
+) -> Result<OperationResult> {
+    if !is_node_process_running(&node_name) {
+        return Ok(OperationResult {
+            success: false,
+            message: format!("Node '{}' is not running.", node_name),
+        });
+    }
+
+    let node_stopped = {
+        let mut manager = state
+            .node_manager
+            .lock()
+            .map_err(|e| AppError::Custom(format!("Failed to lock node manager: {}", e)))?;
+        if let Some(mut node_process) = manager.nodes.remove(&node_name) {
+            if let Some(mut process) = node_process.process.take() {
+                process.kill().is_ok()
+            } else {
+                false
+            }
+        } else {
+            false
+        }
+    };
```

Comment:
another way to write this (since the other paths lead to the same thing) is:

```suggestion
    let node_stopped = 'done: {
        let mut manager = state
            .node_manager
            .lock()
            .map_err(|e| AppError::Custom(format!("Failed to lock node manager: {}", e)))?;
        if let Some(mut node_process) = manager.nodes.remove(&node_name) {
            if let Some(mut process) = node_process.process.take() {
                break 'done process.kill().is_ok();
            }
        }
        false
    };
```